src/mon/FSCommands.cc

```diff
@@ -432,6 +432,12 @@ class SetHandler : public FileSystemCommandHandler
 	if (r != 0) {
 	  return r;
 	}
+        std::bitset<MAX_MDS> rank_mask = std::bitset<MAX_MDS>(bin_string);
+        if (!rank_mask.test(0)) {
+          ss << "the first bit must be set for rank0";
```

Comment:
explain (to the user)?

Reply:
There have been some slight changes in the implementation. My current implementation involves creating a subtree root for the target directory and exporting that node to a different rank. Then, rebalancing is performed within the masked ranks. For example, if you run "setfattr -n ceph.dir.bal.mask -v 1,2 /home/backup," it converts /home/backup into a subtree and moves it to rank 1. Subsequently, the balancer queries the bal_rank_mask vxattr values of subtrees with authority on that rank and redistributes them within the range of 1 to 2. The '/' root inode is indeed fixed at rank 0, making it impossible to move it to a different rank and apply the logic.  Therefore, I’ve added the constraint to ensure that rank 0 is always included for the root ‘/‘.

The existing bal_rank_mask implementation in https://github.com/ceph/ceph/pull/43284 sends unpinned subtrees (excluding the root '/') to masked ranks for redistribution. The ceph.dir.bal.mask functionality has been implemented, and during the process of integrating the bal_rank_mask feature, constraints were added out of necessity.